### DISCUSSION TITLE
Polish Translation

### DISCUSSION THREAD

**Discussion by Haas888** [bafolts/terraforming-mars#2060] [Localization]
Hello,
I have zero experience in github/programming, but I love the game and I would like to contribute by helping with translations to prepare a polish version.

**Comment by vsrisuknimit** [comment]
@Haas888 check out the locales folder (
https://github.com/bafolts/terraforming-mars/tree/main/src/locales). I believe the Chinese/Russian ones are pretty complete in terms of translation. You can use those files as a template, and start translating the output text to Polish. When you are done, let us know and the translation files can be integrated to the main repo.

**Comment by bafolts** [comment]
@Haas888 not sure what your extent of github or programming knowledge is but at a minimum will need to understand these JSON files.

https://github.com/bafolts/terraforming-mars/blob/main/src/locales/de/cards.json

Will need a `cards.json` for polish as well as the other json files in that directory. In this file the english value is the left side value and the translation value is on the right side. If you provide equivalents for polish someone can commit them to the repository for you.

**Comment by Haas888** [comment]
I have finished cards.json file, could you please verify? :)

https://github.com/Haas888/Terraforming-Mars-pl

**Reply by sienmich** [reply]
Nice! I'll take a look soon. Na pierwszy rzut oka wygląda legitnie :)

**Reply by xandi404** [reply]
Hey, how can I replace these files? :D

**Reply by kberg** [reply]
Our Polish translation lives at https://github.com/terraforming-mars/terraforming-mars/tree/main/src/locales/pl

**Comment by sienmich** [comment]
Ok, good work. Looks nice to me.
Did you use translations from polish version of the game or you made your own ones?
I'm not sure if all card's text info will always fit, we'll have to see.
If think it's best to wait until you are done with everything and than I'll upload it? Sounds reasonable?
Thanks for doing this!

**Reply by Haas888** [reply]
I used official polish translations of course, being a perfectionist doesn't allow me to do otherwise.
I figured out that the text is a little longer than in other languages, but that's just beauty of our language, isn't it? :)
Sure, I'll try to finish my work first, just wanted to know if it's correct from the technical side. 
Besides, do you import those translations manually card by card or is it somehow automated and I should keep to some "chronology"? I'm not really keen on programming languages and github.

Dzięki wielkie za feedback!

**Reply by sienmich** [reply]
Yep, it is, good 👍 
The imports are automated, I'll have to just import those files that you produced. Order of translations within one file is irrelevant

**Comment by Haas888** [comment]
I think I have finished all I could for now. I don't personally own Turmoil or Promos so I'm not confident enough to prepare translations on my own.
Should I move my .json files somewhere?

**Reply by alrusdi** [reply]
Attach them to this issue and we will import them to main repository

**Comment by Crysthorn** [comment]
Hi. I do have Turmoil, all promos and basically everything possible (I'm Polish as well). If you need help, I can be of service.

**Reply by Haas888** [reply]
Nice, actually it would be best if you could create your own repository based on the Chinese or Russian one. We could then import all files to the main branch just like they did with mine.
All help I can get is very welcome.

**Reply by Crysthorn** [reply]
Haas888, hop on Discord when you have time - we have a convo established there with Sienmich and one of my friends, we added you there already.

**Comment by kberg** [comment]
This is done.
